File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<int>temp;
    void solve(int i,int n,vector<int>nums,set<vector<int>>&st)
    {
        if(i == n)
        {
            st.insert(temp);
            return;
        }
        temp.push_back(nums[i]);
        solve(i+1,n,nums,st);
        temp.pop_back();
        solve(i+1,n,nums,st);
    }
    
    
    vector<vector<int>> subsets(vector<int>& nums) {
        
        vector<vector<int>>v;
        int n = nums.size();
        set<vector<int>>st;
        solve(0,n,nums,st);
        for(auto x : st)
        {
          v.push_back(x);   
        }
        return v;
        
    }
};

File: 0078-subsets/0078-subsets.py
class Solution(object):
    def subsets(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        ans = []
        temp = []
       
        def solve(idx,nums):
            if idx == len(nums):
                ans.append(temp[:])
                return 
            temp.append(nums[idx])
            solve(idx+1,nums)
            temp.pop()
            solve(idx+1,nums)
        
        
        solve(0,nums)
        return ans
        